//HANDLING UNCAUGHT EXCEPTION (synchronized tasks)
process.on('uncaughtException', err => {
  console.log('UNCAUGHT EXCEPTION! ðŸ’¥ Shutting down...');
  console.log(err.name, err.message);
  process.exit(1);
});

const app = require('./app');

const port = process.env.PORT || 8000;
const server = app.listen(port, () => {
  console.log(`App running on port ${port}...`);
});

process.on('unhandledRejection', err => {
  console.log(err.name, err.message);
  console.log('UNHANDLER REJECTION! ðŸ’¥ Shutting down...');
  // Gives the server time to finish all request that are still pending before shutting down.
  server.close(() => {
    process.exit(1);
  });
});
